### PR TITLE
Update server IP helper functions

### PR DESCRIPTION
**Pull request by ajmyyra**
Server IP helper functions were from a prehistoric era where private networking meant what we currently refer as utility network. Updated functions to use the newer networking attribute in server details, and to also support utility IP addresses with its own helper function.

### CODE REVIEW

**Review by nate-droid — APPROVED** [review]

File: upcloud_api/server.py [hunk 1]
```diff
@@ -505,28 +504,35 @@ def get_ip(self, access='public', addr_family=None, strict=None):
...
+        if not hasattr(self, 'networking'):
+            raise Exception(
+                "`networking` attribute is missing, server details must be fetched first"
+            )
 
-        # server can have several public or private IPs
-        ip_addrs = [ip_addr for ip_addr in self.ip_addresses if ip_addr.access == access]
+        ip_addrs = []
+        for iface in self.networking['interfaces']['interface']:
+            ip_addresses = iface['ip_addresses']['ip_address']
```
Comment (Darep) [inline upcloud_api/server.py:517]: `ip_addrs` and `ip_addresses` in same scope feels a bit funny – what about `iface_ip_addrs` here? 🤔 

```suggestion
            iface_ip_addrs = iface['ip_addresses']['ip_address']
```

File: upcloud_api/server.py [hunk 2]
```diff
@@ -505,28 +504,35 @@ def get_ip(self, access='public', addr_family=None, strict=None):
...
+            if len(ip_addresses) == 0:
+                continue
 
-        # prefer addr_family (or IPv4 if none given)
-        preferred_family = addr_family if addr_family else 'IPv4'
```
Comment (Darep) [inline upcloud_api/server.py:518]: Is this a breaking change? As the get_ip() will behave differently now 🤔 At least should update the method documentation for "ip_addr_family" at the top to mention that None returns all :blush:

Comment (ajmyyra) [reply upcloud_api/server.py:518]: We're still returning IPv6 address if addr_family is not defined. I guess we could make an extra check if we have a v4 address available in the possible IPs so interface ordering does not matter, I'll add that.

File: upcloud_api/server.py (hunk 1 above)
Comment (ajmyyra) [reply upcloud_api/server.py:517]: Sure!

File: upcloud_api/server.py (hunk 2 above)
Comment (Darep) [reply upcloud_api/server.py:518]: Yeah was just thinking if someone has code like `ip_address = get_ip()[0]` and server has 1 ipv4 + 1 ipv6, previously they would get only ipv4 guaranteed, but now i guess they might get list of 2 with ipv6 in front 🤔

File: upcloud_api/server.py (hunk 1 above)
Comment (ajmyyra) [reply upcloud_api/server.py]: Changed.

File: upcloud_api/server.py (hunk 2 above)
Comment (ajmyyra) [reply upcloud_api/server.py:518]: Changed to prefer IPv4 when no IP family has been defined.

**Review by Darep — APPROVED** [review]
LGTM!

**Pull request merged**